server/semantics-service/src/main/java/edu/illinois/ncsa/incore/service/semantics/controllers/TypeController.java

```diff
@@ -109,6 +109,13 @@ public Response listTypes(
 
         List<Document> typeList = this.typeDAO.getTypes();
 
+        // hackish way of adding "id" field
```

Comment:
I don't think we need to do this here. The listTypes returns all the type "names", no objects, right @longshuicy ?

Reply:
If set "detail" flag to true it returns all the objects :-(
We might want to revisit this design at some point...